bundles/org.openhab.binding.wolfsmartset/src/main/java/org/openhab/binding/wolfsmartset/internal/api/WolfSmartsetApi.java

```diff
@@ -0,0 +1,622 @@
+            loginFailedCounter++;
+            logger.info("Error while parsing device descriptions: {}", e.getMessage());
+        }
+        if (systemState != null && systemState.length >= 1)
+            return systemState;
+        else
+            return null;
+    }
+
+    /**
+     * Request the fault messages of the given system
+     * 
+     * @param systemId
+     * @param gatewayId
+     * @return
+     */
+    public @Nullable ReadFaultMessagesDTO getFaultMessages(Integer systemId, Integer gatewayId) {
+        final String response = getFaultMessagesString(systemId, gatewayId);
+        ReadFaultMessagesDTO faultMessages = null;
+        try {
+            faultMessages = GSON.fromJson(response, ReadFaultMessagesDTO.class);
+        } catch (JsonSyntaxException | IllegalStateException | ClassCastException e) {
+            loginFailedCounter++;
+            logger.info("Error while parsing faultmessages: {}", e.getMessage());
+        }
+        return faultMessages;
+    }
+
+    /**
+     * request the current values for a unit associated with the given system.
+     * if lastAccess is not null, only value changes newer than the given timestamp are returned
+     * 
+     * @param systemId
+     * @param gatewayId
+     * @param bundleId the id of the Unit
+     * @param valueIdList list of the values to request
+     * @param lastAccess timestamp of the last valid value request
+     * @return
+     */
+    public @Nullable GetParameterValuesDTO getGetParameterValues(Integer systemId, Integer gatewayId, Long bundleId,
+            List<Long> valueIdList, @Nullable Instant lastAccess) {
+        final String response = getGetParameterValuesString(systemId, gatewayId, bundleId, valueIdList, lastAccess);
+        GetParameterValuesDTO parameterValues = null;
+        try {
+            parameterValues = GSON.fromJson(response, GetParameterValuesDTO.class);
+        } catch (JsonSyntaxException | IllegalStateException | ClassCastException e) {
+            loginFailedCounter++;
+            logger.info("Error while parsing device parameter values: {}", e.getMessage());
+        }
+        return parameterValues;
+    }
+
+    private void startClient() throws WolfSmartsetCloudException {
+        if (!httpClient.isStarted()) {
+            try {
+                httpClient.start();
+            } catch (Exception e) {
+                throw new WolfSmartsetCloudException("No http client cannot be started: " + e.getMessage(), e);
+            }
+        }
+        setDelay(delay);
+    }
+
+    public void stopClient() {
+        try {
+            stopProcessJob();
+            requestQueue.forEach(queueEntry -> queueEntry.future.completeExceptionally(new CancellationException()));
+            this.httpClient.stop();
+        } catch (Exception e) {
+            logger.debug("Error stopping httpclient :{}", e.getMessage(), e);
+        }
+    }
```

Comment:
@daMihe i copied a lot from the miio binding, and this comes from [https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/cloud/MiCloudConnector.java#L119](https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/cloud/MiCloudConnector.java#L119) 

@marcelrv do you know the reason for the httpClient.stop?

Reply:
@marcelrv uses in miio-binding a self maintained httpclient and creates it [here](https://github.com/openhab/openhab-addons/blob/2cc08ba8bd28ae1647bf8e3e2637cb978b2cf3bb/bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/cloud/CloudConnector.java#L94). Whether that is useful there is out of scope here. If you do not need to modify the `httpclient` itself, you can and should use the shared instance. But then you must not modify it.

Reply:
Indeed as @daMihe indicated, it is because I don't use the shared instance.
Normally this should not be required when using the (recommended way) shared httpclient